binary_search_tree/tree.py

```diff
@@ -9,51 +9,119 @@ def __init__(self, key, val = None):
+ 
     def find(self, key):
-        pass
+        current = self.root
+
+        while current:
+            if current.key == key:
+                return current.value
+            elif key < current.key:
+                current = current.left
+            else:
+                current = current.right
+
+        return None
+
+    # Time Complexity: O(log n)
+    # Space Complexity: O(log n)
+
+
+    def inorder_helper_function(self, current, node_values):
+        if not current:
+            return node_values
+
+        self.inorder_helper_function(current.left, node_values)
+        node_values.append({ 
+            "key": current.key,
+            "value": current.value
+        })
+        self.inorder_helper_function(current.right, node_values)
 
-    # Time Complexity: 
-    # Space Complexity: 
+        return node_values
+    
     def inorder(self):
-        pass
+        values = []
+        return self.inorder_helper_function(self.root, values)
+
+    # Time Complexity: O(log n)
+    # Space Complexity: O(log n)
+
+  
+    def preorder_helper_function(self, current, node_values):
+        if not current:
+            return node_values
+
+        node_values.append({ 
+            "key": current.key,
+            "value": current.value
+        })
+        self.preorder_helper_function(current.left, node_values)        
+        self.preorder_helper_function(current.right, node_values)
+
+        return node_values
 
-    # Time Complexity: 
-    # Space Complexity:     
     def preorder(self):
-        pass
+        values = []
+        return self.preorder_helper_function(self.root, values)        
 
-    # Time Complexity: 
-    # Space Complexity:     
-    def postorder(self):
-        pass
+    # Time Complexity: O(log n)
+    # Space Complexity: O(log n)
 
-    # Time Complexity: 
-    # Space Complexity:     
-    def height(self):
-        pass
 
+    def postorder_helper_function(self, current, node_values):
+        if not current:
+            return node_values
+
+        self.postorder_helper_function(current.left, node_values)        
+        self.postorder_helper_function(current.right, node_values)
+        node_values.append({ 
+            "key": current.key,
+            "value": current.value
+        })
 
-#   # Optional Method
-#   # Time Complexity: 
-#   # Space Complexity: 
-    def bfs(self):
-        pass
+        return node_values
 
+    def postorder(self):
+        values = []
+        return self.postorder_helper_function(self.root, values)
         
 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(log n)
+    
+    
+    def height_helper(self, current):
+        if not current:
+            return 0
 
-#   # Useful for printing
-    def to_s(self):
-        return f"{self.inorder()}"
+        return max(self.height_helper(current.left), self.height_helper(current.right)) + 1
+    
+    def height(self):
+        return self.height_helper(self.root)        
```

Comment:
It doesn't seem you added a time or space complexity calculation for your implementation of `height`, but I am not going to move down to a Yellow, but I do encourage you to think through it.